### PR TITLE
8318219: RISC-V: C2 ExpandBits

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you review the change to add intrinsic for ExpandBits for Long & Integer?
Thanks!

related to: https://github.com/openjdk/jdk/pull/16481

### jtreg
test/jdk/java/lang/CompressExpand*.java
test/hotspot/jtreg/compiler/intrinsics/TestBitShuffleOpers.java
test/hotspot/jtreg/compiler/intrinsics/

### gtest
test/hotspot/gtest/opto/test_compress_expand_bits.cpp

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318219](https://bugs.openjdk.org/browse/JDK-8318219): RISC-V: C2 ExpandBits (**Sub-task** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16658/head:pull/16658` \
`$ git checkout pull/16658`

Update a local copy of the PR: \
`$ git checkout pull/16658` \
`$ git pull https://git.openjdk.org/jdk.git pull/16658/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16658`

View PR using the GUI difftool: \
`$ git pr show -t 16658`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16658.diff">https://git.openjdk.org/jdk/pull/16658.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16658#issuecomment-1810488317)

### CODE REVIEW

File: test/hotspot/jtreg/compiler/intrinsics/TestBitShuffleOpers.java [hunk 1]
```diff
@@ -30,7 +30,8 @@
  * @requires (((os.arch=="x86" | os.arch=="amd64" | os.arch=="x86_64") &
  *            (vm.cpu.features ~= ".*bmi2.*" & vm.cpu.features ~= ".*bmi1.*" &
  *             vm.cpu.features ~= ".*sse2.*")) |
- *            (os.arch=="aarch64" & vm.cpu.features ~= ".*svebitperm.*"))
+ *            (os.arch=="aarch64" & vm.cpu.features ~= ".*svebitperm.*") |
+ *            (os.arch=="riscv64" & vm.cpu.features ~= ".*v.*"))
```
[the remark this answers is not in the document]
Comment (Hamlin-Li) [reply test/hotspot/jtreg/compiler/intrinsics/TestBitShuffleOpers.java]: Thanks for pointing to the link!

**Comment by Hamlin-Li** [comment]
Seems the C2_MacroAssembler::expand_bits_v and compress_bits_v could be merged, please let me known if you also think so.

**Review by RealFYang — APPROVED** [review]
Updated change looks good to me assuming that the following modified test could still be matched and test good. Thanks.

**Comment by Hamlin-Li** [comment]
Thanks @RealFYang for your reviewing.
Yes, the test is matched with the new cpu feature string, and passed.

**Pull request closed without merging**